draft-js-inline-toolbar-plugin/src/components/Toolbar/index.js

```diff
@@ -45,25 +63,103 @@ export default class Toolbar extends React.Component {
    */
   onOverrideContent = (overrideContent) => {
     this.setState({ overrideContent });
-  }
+  };
 
   onSelectionChanged = () => {
     // need to wait a tick for window.getSelection() to be accurate
     // when focusing editor with already present selection
     setTimeout(() => {
       if (!this.toolbar) return;
-      const relativeParent = getRelativeParent(this.toolbar.parentElement);
-      const toolbarHeight = this.toolbar.clientHeight;
-      const relativeRect = (relativeParent || document.body).getBoundingClientRect();
+
+      if (typeof this.state.width !== 'number') {
+        this.setState({ width: this.toolbar.offsetWidth });
+      } else {
+        // toolbar width must forcibly overwritten to prevent unexpected geometry
+        // changes
+        this.toolbar.style.width = `${this.state.width}px`;
+      }
+
+      const metrics = {
+        rule: 'left',
+        ruleValue: 0,
+        shift: null,
+        shiftValue: 0
+      };
```

Comment:
this object is fairly confusing imo.

Make them variables instead of an object and define the following:

```
let alignment = 'left';
let horizontalOffset = 0;
```

You shouldn't really need more than these two values. Since shift and rule are both set to either 'left' or 'right'

shiftValue is unused...